include/bluetooth/audio/tbs.h

```diff
@@ -328,4 +328,411 @@ void bt_tbs_register_cb(struct bt_tbs_cb *cbs);
+ * @param inst_index    The index of the TBS instance that was updated.
+ * @param call_index    The call index.
+ * @param reason        The termination reason.
+ */
+typedef void (*bt_tbs_client_termination_reason_cb)(struct bt_conn *conn,
+						    int err, uint8_t inst_index,
+						    uint8_t call_index,
+						    uint8_t reason);
+
+/**
+ * @brief Callback function for ccp_read_current_calls.
+ *
+ * @param conn          The connection used in the function.
+ * @param err           Error value. BT_TBS_CLIENT_RESULT_CODE_*,
+ *                      GATT error or errno value.
+ * @param inst_index    The index of the TBS instance that was updated.
+ * @param call_count    Number of calls read.
+ * @param calls         Array of calls. The array is not kept by
+ *                      the client, so must be copied to be saved.
+ */
+typedef void (*bt_tbs_client_current_calls_cb)(struct bt_conn *conn, int err,
+					       uint8_t inst_index,
+					       uint8_t call_count,
+					       const struct bt_tbs_client_call *calls);
+
+/**
+ * @brief Callback function for ccp_read_call_state.
+ *
+ * @param conn          The connection used in the function.
+ * @param err           Error value. BT_TBS_CLIENT_RESULT_CODE_*,
+ *                      GATT error or errno value.
+ * @param inst_index    The index of the TBS instance that was updated.
+ * @param call_count    Number of call states read.
+ * @param call_states   Array of call states. The array is not kept by
+ *                      the client, so must be copied to be saved.
+ */
+typedef void (*bt_tbs_client_call_states_cb)(struct bt_conn *conn, int err,
+					     uint8_t inst_index,
+					     uint8_t call_count,
+					     const struct bt_tbs_client_call_state *call_states);
+
+struct bt_tbs_client_cb {
+	bt_tbs_client_discover_cb            discover;
+	bt_tbs_client_cp_cb                  originate_call;
+	bt_tbs_client_cp_cb                  terminate_call;
+	bt_tbs_client_cp_cb                  hold_call;
+	bt_tbs_client_cp_cb                  accept_call;
+	bt_tbs_client_cp_cb                  retrieve_call;
+	bt_tbs_client_cp_cb                  join_calls;
+
+	bt_tbs_client_read_string_cb         bearer_provider_name;
+	bt_tbs_client_read_string_cb         bearer_uci;
+	bt_tbs_client_read_value_cb          technology;
+	bt_tbs_client_read_string_cb         uri_list;
+	bt_tbs_client_read_value_cb          signal_strength;
+	bt_tbs_client_read_value_cb          signal_interval;
+	bt_tbs_client_current_calls_cb       current_calls;
+	bt_tbs_client_read_value_cb          ccid;
+	bt_tbs_client_read_value_cb          status_flags;
+	bt_tbs_client_read_string_cb         call_uri;
+	bt_tbs_client_call_states_cb         call_state;
+	bt_tbs_client_read_value_cb          optional_opcodes;
+	bt_tbs_client_termination_reason_cb  termination_reason;
+	bt_tbs_client_read_string_cb         remote_uri;
+	bt_tbs_client_read_string_cb         friendly_name;
```

Comment:
Does it make sense to make the client features configurable via Kconfig to reduce the memory consumption?
Simplest and sufficient for most of the users configuration would be to Accept and Terminate and track Call State.

e.g most of the TBS functionality is not used by implementation that has no display.

Reply:
Absolutely! 
We actually even already have https://github.com/zephyrproject-rtos/zephyr-bt-sig/issues/29 and https://github.com/zephyrproject-rtos/zephyr-bt-sig/issues/28 for such ideas. 

We definitely need to be able to disable anything not mandatory.

Reply:
Great!